modules/framework/galasa-parent/dev.galasa.framework.api.ras/src/main/java/dev/galasa/framework/api/ras/internal/routes/GroupDetailsRoute.java

```diff
@@ -0,0 +1,100 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.regex.Matcher;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import dev.galasa.framework.api.common.HttpRequestContext;
+import dev.galasa.framework.api.common.InternalServletException;
+import dev.galasa.framework.api.common.ResponseBuilder;
+import dev.galasa.framework.api.common.ServletError;
+import dev.galasa.framework.api.ras.internal.common.RunActionJson;
+import dev.galasa.framework.api.ras.internal.common.RunActionStatus;
+import dev.galasa.framework.api.ras.internal.common.RunStatusUpdate;
+import dev.galasa.framework.spi.FrameworkException;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.ResultArchiveStoreException;
+import dev.galasa.framework.spi.rbac.RBACException;
+import dev.galasa.framework.spi.utils.GalasaGson;
+
+public class GroupDetailsRoute extends RunsRoute {
+
+   private IFramework framework;
+   private RunStatusUpdate runStatusUpdate;
+   static final GalasaGson gson = new GalasaGson();
+
+   // Regex to match endpoint: /ras/groups/{groupid}
+   protected static final String path = "\\/groups\\/([A-Za-z0-9.\\-=]+)\\/?";
+
+   public GroupDetailsRoute(ResponseBuilder responseBuilder, IFramework framework) throws RBACException {
+      super(responseBuilder, path, framework);
+      this.framework = framework;
+      this.runStatusUpdate = new RunStatusUpdate(framework);
+   }
+
+   @Override
+   public HttpServletResponse handlePutRequest(String pathInfo, HttpRequestContext requestContext,
+         HttpServletResponse res) throws ServletException, IOException, FrameworkException {
+
+      HttpServletRequest request = requestContext.getRequest();
+      String groupId = getGroupIdFromPath(pathInfo);
+
+      RunActionJson runAction = runStatusUpdate.getUpdatedRunActionFromRequestBody(request);
+      String responseBody = updateRunStatus(groupId, runAction);
+
+      return getResponseBuilder().buildResponse(request, res, "text/plain", responseBody, HttpServletResponse.SC_ACCEPTED);
+   }
+
+   private String getGroupIdFromPath(String pathInfo) throws InternalServletException {
+      Matcher matcher = this.getPathRegex().matcher(pathInfo);
+      matcher.matches();
+      String groupId = matcher.group(1);
+      return groupId;
+   }
+
+   private String updateRunStatus(String groupId, RunActionJson runAction) throws InternalServletException, ResultArchiveStoreException {
+
+      String responseBody = "";
+      List<IRunResult> groupedRuns = getRunsByGroupId(groupId);
+
+      if (groupedRuns != null && !groupedRuns.isEmpty()) {
+
+         for (IRunResult run : groupedRuns) {
+
+            RunActionStatus status = RunActionStatus.getfromString(runAction.getStatus());
+            String result = runAction.getResult();
```

Comment:
Perhaps these two lines should be outside of the loop as it looks like they will be the same each iteration as their values come from the `runAction` which is passed in

Reply:
Just to avoid running these lines multiple times if the value will always be the same.

Reply:
If there is a reason they are in the loop that I am missing let me know :)